### PR TITLE
feat(linter): do not report directive in commonjs files

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Closes https://github.com/biomejs/biome/issues/3710

## Test Plan

Added a new test

### CODE REVIEW

File: crates/biome_js_analyze/tests/specs/suspicious/noRedundantUseStrict/commonJsValid.js.snap.new [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
```
Comment (Conaclos) [inline crates/biome_js_analyze/tests/specs/suspicious/noRedundantUseStrict/commonJsValid.js.snap.new:1]: `cargo insta accept` :)

**Review by Conaclos — COMMENTED** [review]
`noRedundantUseStrict` allows using `"use strict"` directive in a script file.

If I remember correctly you changed how `.js` files are recognized as script or module based on `package.json`'s `type` field?
In this case we should not have to check for `package.json`'s `type` in this linter rule because the file should be recognized as a script?

**Comment by ematipico** [comment]
> `noRedundantUseStrict` allows using `"use strict"` directive in a script file.
> 
> If I remember correctly you changed how `.js` files are recognized as script or module based on `package.json`'s `type` field? In this case we should not have to check for `package.json`'s `type` in this linter rule because the file should be recognized as a script?

Your memory of the analyzer is amazing, you're right!

**Review by Conaclos — APPROVED** [review]

**Comment by anthonyshew** [comment]
Thanks!

**Pull request merged**